pkg/blob/azure.go

```diff
@@ -46,21 +46,22 @@ func IsAzureStackCloud(cloud *azureprovider.Cloud) bool {
 	return strings.EqualFold(cloud.Config.Cloud, "AZURESTACKCLOUD")
 }
 
-// GetCloudProvider get Azure Cloud Provider
-func GetCloudProvider(kubeconfig string) (*azureprovider.Cloud, error) {
+// getCloudProvider get Azure Cloud Provider
+func getCloudProvider(kubeconfig string) (*azureprovider.Cloud, error) {
 	kubeClient, err := getKubeClient(kubeconfig)
 	if err != nil && !os.IsNotExist(err) && err != rest.ErrNotInCluster {
 		return nil, fmt.Errorf("failed to get KubeClient: %v", err)
 	}
 
 	az := &azureprovider.Cloud{}
+
 	if kubeClient != nil {
 		klog.V(2).Infof("reading cloud config from secret")
 		az.KubeClient = kubeClient
 		az.InitializeCloudFromSecret()
 	}
 
-	if az.TenantID == "" || az.SubscriptionID == "" {
+	if az.TenantID == "" || az.SubscriptionID == "" || az.ResourceGroup == "" {
```

Comment:
why add a new condition `az.ResourceGroup == ""` ?

Reply:
In the previous tests, without the field in azure.json, there was an error. After adding it, the driver worked fine.

Reply:
which case failed? that means `TenantID` and `SubscriptionID` is not empty, while `ResourceGroup` is empty.

Reply:
Not the case in our tests. It is in the non-AKS clusters. They used their own custom azure.json to run the driver.